### DISCUSSION TITLE
About  channel activity detection (cad) in sx126x

### DISCUSSION THREAD

**Discussion by yufu186** [Lora-net/LoRaMac-node#1026] [Q&A]
In sx126x.c, SX126xSetCadParams is proposed to set the parameters. However, the meaning of cadDetPeak and cadDetMin is not well explained. The datasheet of semtech only gives a vague explanation about the "sensitivity". 
In https://semtech.my.salesforce.com/sfc/p/#E0000000JelG/a/2R000000Q1ES/SPexo9njbhEQLcJVUg1i0Su8p3tpAtwX1jhMBGlXsQI, semtech evalutes the how to select the best parameters in CAD, but it doesn't give a clear explanation.
So What's the real function and meaning of those two parameters? Otherwise, Where does LoRa perform a correlation calculation in CAD? Can we control the calculation process?

**Comment by smtcbboulet** [comment]
Hello, 

This AN presents the best parameters for CAD and these parameters are based a lot of PER test (done by myself).
The CAD calculation is done during the preamble. We can't control the process, it's done in the IP.

regards,

Ben

**Reply by yufu186** [reply]
Hi，
Thanks for your help and response, I still have some issues:
1. Although we can't control the process, can we know the registers used in CAD and can we access them?
2. For sx1276, the datasheet shows that sx1276 doesn't support function to change the CAD parameters (e.g. cadDetPeak and cadDetMin), so what's the default value of them in sx1276?
3. The final issue is: how to understand "cadDetPeak and cadDetMin", Semtech did not give a clear explanation，what's the unit of them, SNR? or others? In the AN, I observe that with the increasing of cadDetPeak, the CAD Rx(%) decreases, what's the reason of that? 
I really need your help!

Regards,

Fu

**Reply by mluis1** [reply]
On the following forum there are some topics discussing the same kind of questions.

https://forum.lora-developers.semtech.com/

I would recommend to go there in order to get better support on Semtech radios usage.

**Reply by yufu186** [reply]
> On the following forum there are some topics discussing the same kind of questions.
> 
> https://forum.lora-developers.semtech.com/
> 
> I would recommend to go there in order to get better support on Semtech radios usage.

Thanks for your help, I will try it.
